alexandria/core/serializers.py

```diff
@@ -200,6 +200,34 @@ def validate(self, *args, **kwargs):
 
         return validated_data
 
+    def _prepare_multipart(self):
+        """Massage multipart data into jsonapi-compatible form."""
+
+        # Depending on incoming data, the parser converts the request into
+        # a dict or an immutable QueryDict. In the latter case, we cannot
+        # modify the dict anymore to accomodate the multipart -> jsonapi
+        # conversion as needed, thus we need to unlock it.
+        # As nothing bad comes from just leaving it "mutable", we don't
+        # bother cleaning it up after.
+        if hasattr(self.initial_data, "_mutable"):
+            self.initial_data._mutable = True
+
+        if not isinstance(self.initial_data.get("document"), dict):
+            self.initial_data["document"] = {
+                "type": "documents",
+                "id": self.initial_data["document"],
+            }
+
+        if (original := self.initial_data.get("original")) and not isinstance(
+            original, dict
+        ):
+            self.initial_data["original"] = {"type": "files", "id": original}
```

Comment:
Would this get any easier with the help of `FormDataResourceRelatedField`? See https://github.com/inosca/ebau/blob/main/django/camac/relations.py#L5